docs/generated/go.md

```diff
@@ -0,0 +1,201 @@
+---
+bodyclass: docs
+headline: Go Generated Code Guide
+layout: docs
+sidenav: doc-side-quickstart-nav.html
+type: markdown
+---
+
+<p class="lead">This guide gives an overview of code generated with the protoc grpc go plugin.</p>
+
+<div id="toc"></div>
+
+## Compiler Invocation and Results of Code Generation
+This section provides instructions for using the `grpc` plugin to `protoc-gen-go`, when compiling `.proto` files with `protoc`.
+See https://developers.google.com/protocol-buffers/docs/reference/go-generated for detailed instructions on 
+how to use the `protoc` compiler the `protoc-gen-go` plugin.
+
+When `protoc` is invoked with a `--go_out=plugins=grpc:` argument, gRPC stub and service interface code gets generated
+from the `.proto` files specified in the rest of the command. The names of the output files are computed in the same way as though the "plain" Go
+`protoc-gen-go` was used (as per the description in https://developers.google.com/protocol-buffers/docs/reference/go-generated). But additional content
+is added to the generated `.pb.go` files as follows:
+
+Each `service Foo` in the `.proto` file results in a `FooClient interface` and a `FooServer interface` in the `.pb.go` file.
```

Comment:
I'd make this into a bulleted list:

* Each `service Foo` in the `.proto` file results in a `FooClient interface` and a `FooServer interface` in the `.pb.go` file. For example, a declaration of `service Foo` in `bar.proto` results in a `type FooServer interface` and a `type FooClient interface` in `bar.pb.go`.

* Each `rpc` declaration within a `service` definition in the `.proto` files results in a corresponding method on the `Server` and `Client` interfaces. Method arguments and return values differ depending on whether the method is on the client or server interface, and on the type of RPC: unary, client-streaming, server-streaming, or bidi-streaming.